recipes/duckdb/all/conanfile.py

```diff
@@ -159,6 +166,8 @@ def generate(self):
         tc.variables["ENABLE_UBSAN"] = False
         if is_msvc(self) and not self.options.shared:
             tc.preprocessor_definitions["DUCKDB_API"] = ""
+        if Version(self.version) >= "0.10.0" and self.options.platform != "":
+            tc.variables["DDUCKDB_EXPLICIT_PLATFORM"] = self.options.platform
```

Comment:
```suggestion
        if Version(self.version) >= "0.10.0" and cross_building(self):
            tc.variables["DDUCKDB_EXPLICIT_PLATFORM"] = f"{self.settings.os}_{self.settings.arch}"
```

I guess we could avoid that bew option and fill the definition when cross-building only.

In the nightly test we can have a hint about what is used: https://github.com/duckdb/duckdb/blob/16938041baa047e77378aff013acde6a6d863c52/.github/workflows/NightlyTests.yml#L660